nntrainer/src/pooling2d_layer.cpp

```diff
@@ -65,14 +65,23 @@ Tensor Pooling2DLayer::backwarding(Tensor in, int iteration) {
 }
 
 void Pooling2DLayer::copy(std::shared_ptr<Layer> l) {
-  // NYI
-}
+  std::shared_ptr<Pooling2DLayer> from =
+    std::static_pointer_cast<Pooling2DLayer>(l);
 
-int Pooling2DLayer::setSize(int *size,
-                            nntrainer::Pooling2DLayer::PropertyType type) {
-  int status = ML_ERROR_NONE;
-  // NYI
-  return status;
+  this->pooling_type = from->pooling_type;
+
+  for (unsigned int i = 0; i < POOLING2D_DIM; ++i) {
+    this->pooling_size[i] = from->pooling_size[i];
+    this->stride[i] = from->stride[i];
+    this->padding[i] = from->padding[i];
+  }
+
```

Comment:
IMO, it is better to accept `std::shared_ptr<Pooling2dLayer> l` as an argument. 
`Layer` may not have those properties.

Reply:
no. it is called from NeuralNetwork Class which doesn't know the type of layer.

Reply:
I thought it might be better to do `if (isInstanceOf(Pooling2dLayer)) { /* updatePoolingproperty */ }`
As for as I surveyed, there is no persuasive way of checking `instance of` in cpp. think it is good for now 👍 

Thanks for the explanation!